### PR TITLE
Tyler Tomlinson

### PR DESCRIPTION
**Pull request by tylertomlinson**
Finished all stories
Added factory bot for practice (associations within bots)
All test passing with edge cases
Test coverage 100%

### CODE REVIEW

File: app/models/machine.rb [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
   validates_presence_of :location
-  
+
   belongs_to :owner
+  has_many :machine_snacks
+  has_many :snacks, through: :machine_snacks
+
+  def average_snack_price
+    if snacks.empty?
+      0
```
Comment (megstang) [inline app/models/machine.rb:10]: good edge case here!

File: app/views/snacks/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<h1>Snack Info</h1>
+
+<h3> <%= @snack.name %> </h3
+<h3> Price: $<%= @snack.price %> <h3>
+
+  <h3> Locations </h3>
+  <ul>
+  <% @snack.machines.each do |machine| %>
+    <p id="machine_info-<%= machine.id %>">
+      <%= machine.location %> has <%= machine.snack_counter %> different kinds of snacks, with an average price of <%= number_to_currency(machine.average_snack_price)  %>
```
Comment (megstang) [inline app/views/snacks/show.html.erb:10]: nice job with this line 👍

File: spec/features/machines/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    expect(page).to have_content(snack1.price)
+
+    expect(page).to have_content(snack2.name)
+    expect(page).to have_content(snack2.price)
+
+    expect(page).to have_content(snack3.name)
+    expect(page).to have_content(snack3.price)
+
+    expect(page).to have_content(snack4.name)
+    expect(page).to have_content(snack4.price)
```
Comment (megstang) [inline spec/features/machines/show_spec.rb:30]: this would be a good place to use withins as well.

**Review by megstang — COMMENTED** [review]
@tylertomlinson Great job on this! I didn't really have too much feedback for you. Looks really good, and happy to see you trying out factorybot 🤖
